File: app/models/department_item.rb
class DepartmentItem
  include Mongoid::Document

  belongs_to :item
  embedded_in :department

  field :price, type: BigDecimal
  field :quantity, type: Integer
  field :position, type: Integer
  field :days, type: String, default: 'Mon-Fri'
  field :deduction, type: Hash, default: {}

  validates_presence_of :item
end

File: src/app/models/department-item.ts
import { MongoidId } from './mongoid-id';

interface DepartmentItemPartials {
  days: string;
  deduction?: number[];
  price: number;
  quantity: number;
  position: number;
}

export interface DepartmentItemParams extends DepartmentItemPartials {
  item: string;
}

export interface DepartmentItem extends DepartmentItemPartials, MongoidId {
  item_id: { $oid: string };
}
